src/components/Modal/BaseModal.js

```diff
@@ -76,20 +80,25 @@ function BaseModal({
     );
 
     useEffect(() => {
-        Modal.willAlertModalBecomeVisible(isVisible);
-
-        // To handle closing any modal already visible when this modal is mounted, i.e. PopoverReportActionContextMenu
-        Modal.setCloseModal(isVisible ? onClose : null);
-    }, [isVisible, onClose]);
+        isVisibleRef.current = isVisible;
```

Comment:
NAB. Is there a significant gain merging the two useEffects together? I think it's more readable to have the isVisibleRef useEffect separated